### ISSUE TITLE
DNApy can not open .gb file

### ISSUE THREAD

**Issue by openpaul** [mengqvist/DNApy#24]
I have a .gb file that can not be opened with DNApy. It produces this message:
```
python main.py 
Traceback (most recent call last):
  File "main.py", line 280, in open_file
    genbank.gb = genbank.gbobject(all_path) #make a genbank object and read file
  File "/home/paul/DNApy/genbank.py", line 210, in __init__
    self.opengb(filepath)
  File "/home/paul/DNApy/genbank.py", line 226, in opengb
    self.readgb(infile)
  File "/home/paul/DNApy/genbank.py", line 248, in readgb
    self.parse(''.join(line_list)) 
  File "/home/paul/DNApy/genbank.py", line 268, in parse
    self.gbfile['locus']['name'] = m.group(1) #locus id
AttributeError: 'NoneType' object has no attribute 'group'
```

This is the file. it seems valid to me:
```
LOCUS       pRSET-A	2897 bp	DNA	SYN	23-Sep-2013
DEFINITION  pRSET-A
ACCESSION    
KEYWORDS    
SOURCE    
  ORGANISM  other sequences; artificial sequences; vectors.
FEATURES             Location/Qualifiers
     source          1..2897
                     /organism="pRSET-A"
                     /mol_type="other DNA"
     promoter        20..38
                     /label="T7_promoter"
     misc_feature    75..91
                     /label="T7_transl_en_RBS"
     misc_feature    112..129
                     /label="6xHis"
     misc_feature    132..150
                     /label="Xpress_fwd_primer"
     misc_feature    133..165
                     /label="T7_gene10_leader"
     misc_feature    133..165
                     /label="T7_leader"
     misc_feature    169..192
                     /label="Xpress_EK"
     misc_feature    complement(298..316)
                     /label="T7_Terminal_primer"
     terminator      256..384
                     /label="T7_terminator"
     rep_origin      473..779
                     /label="f1_origin"
     promoter        972..1000
                     /label="AmpR_promoter"
     CDS             1042..1902
                     /label="Ampicillin"
                     /gene="Ampicillin"
                     /note="ORF frame 1"
                     /translation="MSIQHFRVALIPFFAAFCLPVFAHPETLVKVKDAEDQLGARVGY
                     IELDLNSGKILESFRPEERFPMMSTFKVLLCGAVLSRIDAGQEQLGRRIHYSQNDLVE
                     YSPVTEKHLTDGMTVRELCSAAITMSDNTAANLLLTTIGGPKELTAFLHNMGDHVTRL
                     DRWEPELNEAIPNDERDTTMPVAMATTLRKLLTGELLTLASRQQLIDWMEADKVAGPL
                     LRSALPAGWFIADKSGAGERGSRGIIAALGPDGKPSRIVVIYTTGSQATMDERNRQIA
                     EIGASLIKHW*"
     gene            1042..1902
                     /label="Ampicillin"
                     /gene="Ampicillin"
     rep_origin      2057..2676
                     /label="pBR322_origin"
ORIGIN
    1 GATCTCGATC CCGCGAAATT AATACGACTC ACTATAGGGA GACCACAACG GTTTCCCTCT 
   61 AGAAATAATT TTGTTTAACT TTAAGAAGGA GATATACATA TGCGGGGTTC TCATCATCAT 
  121 CATCATCATG GTATGGCTAG CATGACTGGT GGACAGCAAA TGGGTCGGGA TCTGTACGAC 
  181 GATGACGATA AGGATCGATG GGGATCCGAG CTCGAGATCT GCAGCTGGTA CCATGGAATT 
  241 CGAAGCTTGA TCCGGCTGCT AACAAAGCCC GAAAGGAAGC TGAGTTGGCT GCTGCCACCG 
  301 CTGAGCAATA ACTAGCATAA CCCCTTGGGG CCTCTAAACG GGTCTTGAGG GGTTTTTTGC 
  361 TGAAAGGAGG AACTATATCC GGATCTGGCG TAATAGCGAA GAGGCCCGCA CCGATCGCCC 
  421 TTCCCAACAG TTGCGCAGCC TGAATGGCGA ATGGGACGCG CCCTGTAGCG GCGCATTAAG 
  481 CGCGGCGGGT GTGGTGGTTA CGCGCAGCGT GACCGCTACA CTTGCCAGCG CCCTAGCGCC 
  541 CGCTCCTTTC GCTTTCTTCC CTTCCTTTCT CGCCACGTTC GCCGGCTTTC CCCGTCAAGC 
  601 TCTAAATCGG GGGCTCCCTT TAGGGTTCCG ATTTAGTGCT TTACGGCACC TCGACCCCAA 
  661 AAAACTTGAT TAGGGTGATG GTTCACGTAG TGGGCCATCG CCCTGATAGA CGGTTTTTCG 
  721 CCCTTTGACG TTGGAGTCCA CGTTCTTTAA TAGTGGACTC TTGTTCCAAA CTGGAACAAC 
  781 ACTCAACCCT ATCTCGGTCT ATTCTTTTGA TTTATAAGGG ATTTTGCCGA TTTCGGCCTA 
  841 TTGGTTAAAA AATGAGCTGA TTTAACAAAA ATTTAACGCG AATTTTAACA AAATATTAAC 
  901 GCTTACAATT TAGGTGGCAC TTTTCGGGGA AATGTGCGCG GAACCCCTAT TTGTTTATTT 
  961 TTCTAAATAC ATTCAAATAT GTATCCGCTC ATGAGACAAT AACCCTGATA AATGCTTCAA 
 1021 TAATATTGAA AAAGGAAGAG TATGAGTATT CAACATTTCC GTGTCGCCCT TATTCCCTTT 
 1081 TTTGCGGCAT TTTGCCTTCC TGTTTTTGCT CACCCAGAAA CGCTGGTGAA AGTAAAAGAT 
 1141 GCTGAAGATC AGTTGGGTGC ACGAGTGGGT TACATCGAAC TGGATCTCAA CAGCGGTAAG 
 1201 ATCCTTGAGA GTTTTCGCCC CGAAGAACGT TTTCCAATGA TGAGCACTTT TAAAGTTCTG 
 1261 CTATGTGGCG CGGTATTATC CCGTATTGAC GCCGGGCAAG AGCAACTCGG TCGCCGCATA 
 1321 CACTATTCTC AGAATGACTT GGTTGAGTAC TCACCAGTCA CAGAAAAGCA TCTTACGGAT 
 1381 GGCATGACAG TAAGAGAATT ATGCAGTGCT GCCATAACCA TGAGTGATAA CACTGCGGCC 
 1441 AACTTACTTC TGACAACGAT CGGAGGACCG AAGGAGCTAA CCGCTTTTTT GCACAACATG 
 1501 GGGGATCATG TAACTCGCCT TGATCGTTGG GAACCGGAGC TGAATGAAGC CATACCAAAC 
 1561 GACGAGCGTG ACACCACGAT GCCTGTAGCA ATGGCAACAA CGTTGCGCAA ACTATTAACT 
 1621 GGCGAACTAC TTACTCTAGC TTCCCGGCAA CAATTAATAG ACTGGATGGA GGCGGATAAA 
 1681 GTTGCAGGAC CACTTCTGCG CTCGGCCCTT CCGGCTGGCT GGTTTATTGC TGATAAATCT 
 1741 GGAGCCGGTG AGCGTGGGTC TCGCGGTATC ATTGCAGCAC TGGGGCCAGA TGGTAAGCCC 
 1801 TCCCGTATCG TAGTTATCTA CACGACGGGG AGTCAGGCAA CTATGGATGA ACGAAATAGA 
 1861 CAGATCGCTG AGATAGGTGC CTCACTGATT AAGCATTGGT AACTGTCAGA CCAAGTTTAC 
 1921 TCATATATAC TTTAGATTGA TTTAAAACTT CATTTTTAAT TTAAAAGGAT CTAGGTGAAG 
 1981 ATCCTTTTTG ATAATCTCAT GACCAAAATC CCTTAACGTG AGTTTTCGTT CCACTGAGCG 
 2041 TCAGACCCCG TAGAAAAGAT CAAAGGATCT TCTTGAGATC CTTTTTTTCT GCGCGTAATC 
 2101 TGCTGCTTGC AAACAAAAAA ACCACCGCTA CCAGCGGTGG TTTGTTTGCC GGATCAAGAG 
 2161 CTACCAACTC TTTTTCCGAA GGTAACTGGC TTCAGCAGAG CGCAGATACC AAATACTGTT 
 2221 CTTCTAGTGT AGCCGTAGTT AGGCCACCAC TTCAAGAACT CTGTAGCACC GCCTACATAC 
 2281 CTCGCTCTGC TAATCCTGTT ACCAGTGGCT GCTGCCAGTG GCGATAAGTC GTGTCTTACC 
 2341 GGGTTGGACT CAAGACGATA GTTACCGGAT AAGGCGCAGC GGTCGGGCTG AACGGGGGGT 
 2401 TCGTGCACAC AGCCCAGCTT GGAGCGAACG ACCTACACCG AACTGAGATA CCTACAGCGT 
 2461 GAGCTATGAG AAAGCGCCAC GCTTCCCGAA GGGAGAAAGG CGGACAGGTA TCCGGTAAGC 
 2521 GGCAGGGTCG GAACAGGAGA GCGCACGAGG GAGCTTCCAG GGGGAAACGC CTGGTATCTT 
 2581 TATAGTCCTG TCGGGTTTCG CCACCTCTGA CTTGAGCGTC GATTTTTGTG ATGCTCGTCA 
 2641 GGGGGGCGGA GCCTATGGAA AAACGCCAGC AACGCGGCCT TTTTACGGTT CCTGGCCTTT 
 2701 TGCTGGCCTT TTGCTCACAT GTTCTTTCCT GCGTTATCCC CTGATTCTGT GGATAACCGT 
 2761 ATTACCGCCT TTGAGTGAGC TGATACCGCT CGCCGCAGCC GAACGACCGA GCGCAGCGAG 
 2821 TCAGTGAGCG AGGAAGCGGA AGAGCGCCCA ATACGCAAAC CGCCTCTCCC CGCGCGTTGG 
 2881 CCGATTCATT AATGCAG
//

```

**Comment by openpaul** [comment]
I figured out that one problem is the missing "linear/circular" keyword in the LOCUS line.

SnapGene asks me if it is linear or circular an then opens without complain. Will look further into it.

**Comment by 0b0bby0** [comment]
Glad you localized it. I'll get the parsing fixed.

**Comment by openpaul** [comment]
I once more had a file that was not opened by DNApy

I get the feeling, that the parser is a very compley party of the software.
I understand, that you do not want dependecies, but biopython has a very good genbank parser and can read other files too.

I somehow think it would be a lot more easy to use this library for this project.

**Comment by mengqvist** [comment]
Yes, I've noticed that it is not robust in the way it is handling incomplete/erronious genbank files. I've used the parser to open and fetch information from several hundred thousand genbank files from  NCBI for my work. So I'm quite sure that it's working as intended. But yes, when there is missing info, or the file does otherwise not comply with a "proper" genbank file then it does not work. I will probably need to fix this because many of the vector editing programs do not make "proper" genbank files.
